cdap-ui/app/cdap/components/Reports/ReportsList/index.js

```diff
@@ -47,6 +49,16 @@ class ReportsListView extends Component {
       .subscribe(listReports);
   }
 
+  componentWillReceiveProps(nextProps) {
+    // If switching to a different page, stop reports polling on
+    // current page, and start polling on new page
+    if (nextProps.offset !== this.props.offset && this.interval$) {
+      this.interval$.unsubscribe();
```

Comment:
nit: May be make this as a function? Easier to maintain .. `pollForNewPage` or something that makes sense.

Reply:
just as a discussion (shouldn't be done in this PR), we should also think about a pattern we can use to poll from API using ActionCreator and make use of the socket caching...